### ISSUE TITLE
`result_large_err` does not apply to closures

### ISSUE THREAD

**Issue by narpfel** [rust-lang/rust-clippy#16249]
### Summary

The `result_large_err` lint does not apply to closures.

### Lint Name

result_large_err

### Reproducer

I tried this code:

```rust
#![allow(unused)]

pub type Large = [u8; 1024];

pub fn f() -> Result<(), Large> {
    Ok(())
}

pub fn main() {
    let closure = || -> Result<(), Large> {
        Ok(())
    };
}
```

I expected to see this happen: The lint should be triggered by `f` and by `closure`.

Instead, this happened: The lint is only triggered by `f`.

Godbolt link: https://godbolt.org/z/s99dsq9WK

### Version

```text
rustc 1.94.0-nightly (21ff67df1 2025-12-15)
binary: rustc
commit-hash: 21ff67df15329dd7548ccba54b6c6ae9a562124f
commit-date: 2025-12-15
host: x86_64-unknown-linux-gnu
release: 1.94.0-nightly
LLVM version: 21.1.8
```
